atom/renderer/api/atom_api_spell_check_client.cc

```diff
@@ -46,22 +47,23 @@ class SpellCheckClient::SpellcheckRequest {
   using WordMap =
       std::map<base::string16, std::vector<blink::WebTextCheckingResult>>;
 
-  SpellcheckRequest(const base::string16& text,
-                    blink::WebTextCheckingCompletion* completion)
-      : text_(text), completion_(completion) {
-    DCHECK(completion);
-  }
+  SpellcheckRequest(
+      const base::string16& text,
+      std::unique_ptr<blink::WebTextCheckingCompletion> completion)
+      : text_(text), completion_(std::move(completion)) {}
   ~SpellcheckRequest() {}
 
   const base::string16& text() const { return text_; }
-  blink::WebTextCheckingCompletion* completion() { return completion_; }
+  std::shared_ptr<blink::WebTextCheckingCompletion> completion() {
+    return completion_;
+  }
   WordMap& wordmap() { return word_map_; }
 
  private:
   base::string16 text_;  // Text to be checked in this task.
   WordMap word_map_;     // WordMap to hold distinct words in text
   // The interface to send the misspelled ranges to WebKit.
-  blink::WebTextCheckingCompletion* completion_;
+  std::shared_ptr<blink::WebTextCheckingCompletion> completion_;
```

Comment:
is this intentionally a `shared_ptr` rather than a `unique_ptr`? Looks like the commit you linked has a change in shell that uses `unique_ptr` in a similar context.

Reply:
We can use a `unique_ptr`, and `SpellCheckRequest::completion()` can return a raw pointer.

Reply:
Please prefer that, avoid `shared_ptr` where possible.